TCAT/Controllers/HomeOptionsCardViewController+Extensions.swift

```diff
@@ -0,0 +1,353 @@
+                else { return UITableViewCell() }
+            cell.configure(for: .seeAllStops)
+            return cell
+        }
+            // Favorites (including Add First Favorite!), Recent Searches
+        else {
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Cells.placeIdentifier) as? PlaceTableViewCell
+                else { return UITableViewCell() }
+            cell.configure(for: sections[indexPath.section].getItems()[indexPath.row])
+            return cell
+        }
+    }
+}
+
+// MARK: TableView Delegate
+extension HomeOptionsCardViewController: UITableViewDelegate {
+    func scrollViewWillBeginDragging(_ scrollView: UIScrollView) {
+        if let searchBarText = searchBar.text,
+            searchBarText.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty {
+            searchBar.placeholder = Constants.General.searchPlaceholder
+            searchBar.endEditing(true)
+            searchBar.text = nil
+        }
+        searchBar.setShowsCancelButton(false, animated: true)
+        animateInInfoButton()
+    }
+
+    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        return 0
+    }
+
+    func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
+        return nil
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        switch sections[section] {
+        case .favorites, .recentSearches: return headerHeight
+        case .seeAllStops: return HeaderView.separatorViewHeight
+        default: return 0
+        }
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        var header: HeaderView!
+
+        switch sections[section] {
+        case .recentSearches:
+            header = HeaderView(labelText: Constants.TableHeaders.recentSearches, buttonType: .clear, separatorVisible: true, delegate: self)
+        case .favorites:
+            header = HeaderView(labelText: Constants.TableHeaders.favoriteDestinations, buttonType: .add, delegate: self)
+        case .seeAllStops:
+            header = HeaderView(separatorVisible: true)
+        case .searchResults:
+            return nil
+        default: break
+        }
+
+        return header
+    }
+
+    func tableView(_ tableView: UITableView, editingStyleForRowAt indexPath: IndexPath) -> UITableViewCell.EditingStyle {
+        let section = sections[indexPath.section]
+        switch section {
+        case .favorites:
+            if !section.isEmpty, section.getItems()[0].name != Constants.General.firstFavorite {
+                return .delete
+            } else {
+                return .none
+            }
+        case .recentSearches: return .delete
+        default: return .none
+        }
```

Comment:
why not something like this:

```
if (section == .delete) || (section == .favorites && ...) {
  return .delete
} 
return .none
```

Reply:
I can't directly use '==' with just the section types because the enums have associated values; I'd need to create an actual Section.recentSearches with an associated value and *then* compare it, but at that point it would be comparing the values too and not just the types. Could write a function called .isEqualType(to: ) but idk if that really helps much. Opinions? cc @kevinchan159

Reply:
i think you can do `if (case .delete == section) || (case .favorites == section && ...) { }`

Reply:
I can't get it to work :(